server/controllers/finance/feeCenter.js

```diff
@@ -0,0 +1,285 @@
+  var filter = '', ordering = 'ORDER BY fc.label;';
+
+  if (req.query.full === '1') {
+    sql =
+      `
+        SELECT 
+	        fc.id, fc.label, fc.project_id, fc.is_cost, fc.note, fc.is_principal, p.name, p.abbr, p.enterprise_id, p.zs_id, tt.value
+        FROM
+	        fee_center AS fc
+        JOIN
+	        project AS p ON fc.project_id = p.id
+        LEFT JOIN
+	        (
+		        SELECT 
+			        IFNULL(IF(t.is_cost = 1, SUM(t.debit_equiv - t.credit_equiv), SUM(t.credit_equiv - t.debit_equiv)), 0) AS value, t.id 
+            FROM 
+			        (
+				        SELECT 
+					        gl.debit_equiv, gl.credit_equiv, f.is_cost, f.id 
+				        FROM 
+					        general_ledger AS gl 
+				        JOIN
+					        fee_center AS f ON gl.fc_id = f.id 
+			        ) AS t
+	        ) AS tt ON tt.id = fc.id
+      `;
+  }
+
+
+  if(req.query.available === '1') {
+    filter += ' fc.id NOT IN (SELECT s.cc_id FROM service AS s WHERE NOT ISNULL(s.cc_id)) AND fc.id NOT IN (SELECT s.pc_id FROM service AS s WHERE NOT ISNULL(s.pc_id)) ';
+  }
+
+  if(req.query.is_cost){
+    if(filter !== ''){
+      filter += ' AND ';
+    }
+    filter += `fc.is_cost = ${req.query.is_cost} `;
+  }
+
+  if(req.query.is_principal){
+    if(filter !== ''){
+      filter += ' AND ';
+    }
+    filter += `fc.is_principal = ${req.query.is_cost} `;
+  }
+
+  sql = filter === '' ? sql + ordering : sql + 'WHERE ' + filter + ordering;
+
+  db.exec(sql)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(function (err) {
+    console.log('******* erreur', err);
+  })
+  .done();
+}
+
+/**
+* Create a fee center in the database
+*
+* @param {object} req The express request object
+* @param {object} res The express response object
+* @param {object} next The express object to pass the control to the next middleware
+*
+* @example
+* // POST /fee_centers : Insert a fee center
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.create(req, res, next);
+*/
+
+function create (req, res, next) {
+  'use strict';
+
+  var record = req.body;
+  var createFeeCenterQuery = 'INSERT INTO fee_center SET ?';
+
+  delete record.id;
+
+  db.exec(createFeeCenterQuery, [record])
+    .then(function (result) {
+      res.status(201).json({ id: result.insertId });
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+* Update a fee center in the database
+*
+* @param {object} req The express request object
+* @param {object} res The express response object
+* @param {object} next The express object to pass the control to the next middleware
+*
+* @example
+* // PUT /fee_centers : update a fee center
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.update(req, res, next);
+*/
+
+function update (req, res, next) {
+  'use strict';
+
+  var queryData = req.body;
+  var feeCenterId = req.params.id;
+  var updateFeeCenterQuery = 'UPDATE fee_center SET ? WHERE id = ?';
+
+  delete queryData.id;
+
+  lookupFeeCenter(feeCenterId)
```

Comment:
This used to be best practices until we created our `errors/*.js` libraries for error handling.  You can now just do a UPDATE and then throw `NotFound()` if the `affectedRows` is 0 --- less database queries = easier for raspberry pi!